src/main/java/edu/kis/powp/jobs2d/command/ImmutableComplexCommand.java

```diff
@@ -10,8 +10,16 @@
 public class ImmutableComplexCommand implements ICompoundCommand {
     private List<DriverCommand> driverCommandList;
 
-    public ImmutableComplexCommand(List<DriverCommand> driverCommandList) {
-        this.driverCommandList = Collections.unmodifiableList(driverCommandList);
+    public ImmutableComplexCommand(List<DriverCommand> driverCommandList){
+        List<DriverCommand> commands = new ArrayList<>();
+        for (DriverCommand command : driverCommandList) {
+            try {
+                commands.add(command.clone());
+            } catch (CloneNotSupportedException e) {
+                e.printStackTrace();
```

Comment:
not really a nice way to handle a lack of cloning. 
It would be better to add a command rather than ignore it or throw another exception (since there is a risk of mutability).

Reply:
Done, but is it really a good solution? If there is a problem with cloning, a potentially mutable element will be added to the list even though we wanted to avoid it.

Reply:
I agree. The propagation of an exception (e.g., ImmutableCommandCreationException) would be better.